subsys/usb/usb_descriptor.c

```diff
@@ -818,8 +819,48 @@ void ascii7_to_utf16le(int idx_max, int asci_idx_max, u8_t *buf)
 	}
 }
 
+/*
+ * default USB SN string descriptor is CONFIG_USB_DEVICE_SN, but sometimes
+ * we want use another string as SN descriptor such as the chip's unique
+ * ID. So user can implement this function return a string thats will be
+ * replaced the default SN.
+ * please note that the new SN descriptor you changed must has same length
+ * as CONFIG_USB_DEVICE_SN.
+ */
+__weak u8_t *usb_set_runtime_ascii_sn_string_desc(void)
```

Comment:
You can do it that way, but name of the function can be more generic, not limited to usb, something like SoC/board serial number API (cc @carlescufi)

Reply:
BTW what is the use case for this? I understand the same could be done with configuration option.

Reply:
@finikorg if two same product has same VID and PID, we can't distinguish them. So in that case we need give them a different SN. Many MCU has unique ID, and we can generate USB SN from it's UID at runtime.

There should has a way to rewrite SN descriptor, get SN from EEPROM or generated from UID. Regardless of the method, it needs to be implemented this function.

Reply:
@jfischer-phytec-iot This function just change usb's SN string descriptor, not do other work. what's you idea about the function's name?

Reply:
I think about something generic, get_sn_board or get_sn_soc, is not necessarily limited to usb. Fixes: #6593